Number of dephell tests fail in dephell_setuptools.read_setup

When build a package for dephell 0.7.9 in openSUSE (with dephell_setuptools 0.1.2), I get number of test failing:

```
[   16s] =================================== FAILURES ===================================
[   16s] ____________________________ test_dotted_setup_call ____________________________
[   16s] 
[   16s] temp_path = PosixPath('/tmp/pytest-of-abuild/pytest-2/test_dotted_setup_call0')
[   16s] 
[   16s]     def test_dotted_setup_call(temp_path: Path):
[   16s]         path = temp_path / 'setup.py'
[   16s]         path.write_text(dedent("""
[   16s]             import setuptools
[   16s]             setuptools.setup(name='foo')
[   16s]             """))
[   16s] >       root = SetupPyConverter().load(path)
[   16s] 
[   16s] tests/test_converters/test_setuppy.py:39: 
[   16s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   16s] dephell/converters/setuppy.py:76: in load
[   16s]     data = read_setup(path=path, error_handler=logger.debug)
[   16s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   16s] 
[   16s]     def read_setup(*,
[   16s]                    path: Union[str, Path],
[   16s]                    error_handler: Callable[[Exception], Any] = logger.exception,
[   16s]                    readers: Iterable = ALL_READERS):
[   16s]         result = dict()
[   16s]         for reader in readers:
[   16s]             try:
[   16s]                 content = reader(path=path).content
[   16s]             except Exception as e:
[   16s]                 error_handler(e)
[   16s]             else:
[   16s] >               result.update(content)
[   16s] E               TypeError: 'NoneType' object is not iterable
[   16s] 
[   16s] /usr/lib/python3.7/site-packages/dephell_setuptools/_manager.py:33: TypeError
[   16s] ____________________________ test_return_setup_call ____________________________
[   16s] 
[   16s] temp_path = PosixPath('/tmp/pytest-of-abuild/pytest-2/test_return_setup_call0')
[   16s] 
[   16s]     def test_return_setup_call(temp_path: Path):
[   16s]         path = temp_path / 'setup.py'
[   16s]         path.write_text(dedent("""
[   16s]             from setuptools import setup
[   16s]             def main():
[   16s]                 return setup(name='foo')
[   16s]     
[   16s]             if __name__ == '__main__':
[   16s]                 main()
[   16s]             """))
[   16s] >       root = SetupPyConverter().load(path)
[   16s] 
[   16s] tests/test_converters/test_setuppy.py:53: 
[   16s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   16s] dephell/converters/setuppy.py:76: in load
[   16s]     data = read_setup(path=path, error_handler=logger.debug)
[   16s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   16s] 
[   16s]     def read_setup(*,
[   16s]                    path: Union[str, Path],
[   16s]                    error_handler: Callable[[Exception], Any] = logger.exception,
[   16s]                    readers: Iterable = ALL_READERS):
[   16s]         result = dict()
[   16s]         for reader in readers:
[   16s]             try:
[   16s]                 content = reader(path=path).content
[   16s]             except Exception as e:
[   16s]                 error_handler(e)
[   16s]             else:
[   16s] >               result.update(content)
[   16s] E               TypeError: 'NoneType' object is not iterable
[   16s] 
[   16s] /usr/lib/python3.7/site-packages/dephell_setuptools/_manager.py:33: TypeError
[   16s] ___________________________ test_run_setup_function ____________________________
[   16s] 
[   16s] temp_path = PosixPath('/tmp/pytest-of-abuild/pytest-2/test_run_setup_function0')
[   16s] 
[   16s]     def test_run_setup_function(temp_path: Path):
[   16s]         path = temp_path / 'setup.py'
[   16s]         path.write_text(dedent("""
[   16s]             from setuptools import setup
[   16s]             def run_setup():
[   16s]                 return setup(name='foo')
[   16s]     
[   16s]             if __name__ == '__main__':
[   16s]                 run_setup()
[   16s]             """))
[   16s] >       root = SetupPyConverter().load(path)
[   16s] 
[   16s] tests/test_converters/test_setuppy.py:67: 
[   16s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   16s] dephell/converters/setuppy.py:76: in load
[   16s]     data = read_setup(path=path, error_handler=logger.debug)
[   16s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   16s] 
[   16s]     def read_setup(*,
[   16s]                    path: Union[str, Path],
[   16s]                    error_handler: Callable[[Exception], Any] = logger.exception,
[   16s]                    readers: Iterable = ALL_READERS):
[   16s]         result = dict()
[   16s]         for reader in readers:
[   16s]             try:
[   16s]                 content = reader(path=path).content
[   16s]             except Exception as e:
[   16s]                 error_handler(e)
[   16s]             else:
[   16s] >               result.update(content)
[   16s] E               TypeError: 'NoneType' object is not iterable
[   16s] 
[   16s] /usr/lib/python3.7/site-packages/dephell_setuptools/_manager.py:33: TypeError
[   16s] _______________________________ test_dumps_deps ________________________________
[   16s] 
[   16s]     def test_dumps_deps():
[   16s]         path = Path('tests') / 'requirements' / 'setup.py'
[   16s]         converter = SetupPyConverter()
[   16s]         resolver = converter.load_resolver(path)
[   16s]         reqs = Requirement.from_graph(graph=resolver.graph, lock=False)
[   16s]         assert len(reqs) > 2
[   16s]     
[   16s]         content = converter.dumps(reqs=reqs, project=resolver.graph.metainfo)
[   16s]         print(content)
[   16s] >       root = SetupPyConverter().loads(content)
[   16s] 
[   16s] tests/test_converters/test_setuppy.py:93: 
[   16s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   16s] dephell/converters/base.py:42: in loads
[   16s]     root = self.load(path=stream.name)
[   16s] dephell/converters/setuppy.py:76: in load
[   16s]     data = read_setup(path=path, error_handler=logger.debug)
[   16s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   16s] 
[   16s]     def read_setup(*,
[   16s]                    path: Union[str, Path],
[   16s]                    error_handler: Callable[[Exception], Any] = logger.exception,
[   16s]                    readers: Iterable = ALL_READERS):
[   16s]         result = dict()
[   16s]         for reader in readers:
[   16s]             try:
[   16s]                 content = reader(path=path).content
[   16s]             except Exception as e:
[   16s]                 error_handler(e)
[   16s]             else:
[   16s] >               result.update(content)
[   16s] E               TypeError: 'NoneType' object is not iterable
[   16s] 
[   16s] /usr/lib/python3.7/site-packages/dephell_setuptools/_manager.py:33: TypeError
[   16s] ----------------------------- Captured stdout call -----------------------------
[   16s] 
[   16s] # -*- coding: utf-8 -*-
[   16s] 
[   16s] # DO NOT EDIT THIS FILE!
[   16s] # This file has been autogenerated by dephell <3
[   16s] # https://github.com/dephell/dephell
[   16s] 
[   16s] try:
[   16s]     from setuptools import setup
[   16s] except ImportError:
[   16s]     from distutils.core import setup
[   16s] 
[   16s] readme = ''
[   16s] 
[   16s] setup(
[   16s]     long_description=readme,
[   16s]     name='dephell',
[   16s]     version='0.2.0',
[   16s]     description='Dependency resolution for Python',
[   16s]     url='https://dephell.org/',
[   16s]     project_urls={"home": "https://dephell.org/"},
[   16s]     author='orsinium',
[   16s]     license='MIT',
[   16s]     keywords='sample setuptools development',
[   16s]     classifiers=['Development Status :: 3 - Alpha', 'Intended Audience :: Developers', 'Topic :: Software Development :: Build Tools', 'License :: OSI Approved :: MIT License'],
[   16s]     packages=['dephell', 'dephell.actions', 'dephell.commands', 'dephell.config', 'dephell.controllers', 'dephell.converters', 'dephell.models', 'dephell.repositories', 'dephell.repositories._conda', 'dephell.repositories._git', 'dephell.repositories._warehouse'],
[   16s]     package_dir={"": ""},
[   16s]     package_data={"dephell": ["templates/*.j2", "templates/*.sh"]},
[   16s]     install_requires=['attrs', 'cached-property', 'libtest', 'packaging', 'requests'],
[   16s]     extras_require={"windows": ["colorama"]},
[   16s] )
[   16s] 
[   16s] __________________________ test_dependency_links_load __________________________
[   16s] 
[   16s]     def test_dependency_links_load():
[   16s]         content = """
[   16s]             __import__("setuptools").setup(
[   16s]                 name="lol",
[   16s]                 version="0.1.0",
[   16s]                 install_requires=["libtest"],
[   16s]                 dependency_links=["git+https://github.com/gwtwod/poetrylibtest#egg=libtest-0.1.0"],
[   16s]             )
[   16s]         """
[   16s]         converter = SetupPyConverter()
[   16s] >       resolver = converter.loads_resolver(content.strip())
[   16s] 
[   16s] tests/test_converters/test_setuppy.py:108: 
[   16s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   16s] dephell/converters/base.py:90: in loads_resolver
[   16s]     root = self.loads(content)
[   16s] dephell/converters/base.py:42: in loads
[   16s]     root = self.load(path=stream.name)
[   16s] dephell/converters/setuppy.py:76: in load
[   16s]     data = read_setup(path=path, error_handler=logger.debug)
[   16s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   16s] 
[   16s]     def read_setup(*,
[   16s]                    path: Union[str, Path],
[   16s]                    error_handler: Callable[[Exception], Any] = logger.exception,
[   16s]                    readers: Iterable = ALL_READERS):
[   16s]         result = dict()
[   16s]         for reader in readers:
[   16s]             try:
[   16s]                 content = reader(path=path).content
[   16s]             except Exception as e:
[   16s]                 error_handler(e)
[   16s]             else:
[   16s] >               result.update(content)
[   16s] E               TypeError: 'NoneType' object is not iterable
[   16s] 
[   16s] /usr/lib/python3.7/site-packages/dephell_setuptools/_manager.py:33: TypeError
[   16s] =============================== warnings summary ===============================
[   16s] /usr/lib/python3.7/site-packages/html5lib/_trie/_base.py:3
[   16s]   /usr/lib/python3.7/site-packages/html5lib/_trie/_base.py:3: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
[   16s]     from collections import Mapping
[   16s] 
[   16s] /usr/lib/python3.7/site-packages/flatdict.py:18
[   16s]   /usr/lib/python3.7/site-packages/flatdict.py:18: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
[   16s]     class FlatDict(collections.MutableMapping):
[   16s] 
[   16s] /usr/lib/python3.7/site-packages/_pytest/mark/structures.py:325
[   16s]   /usr/lib/python3.7/site-packages/_pytest/mark/structures.py:325: PytestUnknownMarkWarning: Unknown pytest.mark.allow_hosts - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
[   16s]     PytestUnknownMarkWarning,
[   16s] 
[   16s] tests/test_actions/test_dotenv.py::test_read_dotenv[lines18-expected18]
[   16s]   /home/abuild/rpmbuild/BUILD/dephell-0.7.9/dephell/actions/_dotenv.py:38: DeprecationWarning: invalid escape sequence '\o'
[   16s]     value = decode(value, 'unicode-escape')
[   16s] 
[   16s] -- Docs: https://docs.pytest.org/en/latest/warnings.html
[   16s] ===== 5 failed, 177 passed, 3 skipped, 73 deselected, 4 warnings in 11.42s =====
```

[Complete build logs with all details](https://github.com/dephell/dephell_setuptools/files/3908588/_log.txt)

OK, with ``--no-network`` I get only one failing test:
```
[   55s] =================================== FAILURES ===================================
[   55s] __________________________ test_inspect_venv_command ___________________________
[   55s] 
[   55s] temp_path = PosixPath('/tmp/pytest-of-abuild/pytest-0/test_inspect_venv_command0')
[   55s] capsys = <_pytest.capture.CaptureFixture object at 0x7f9f2750c198>
[   55s] 
[   55s]     def test_inspect_venv_command(temp_path: Path, capsys):
[   55s]         venv = VEnv(path=temp_path)
[   55s]         venv.create(python_path=sys.executable)
[   55s]     
[   55s]         config = Config()
[   55s]         config.attach({
[   55s]             'project': str(temp_path),
[   55s]             'venv': str(temp_path),
[   55s]             'nocolors': True,
[   55s]         })
[   55s]     
[   55s]         command = InspectVenvCommand(argv=[], config=config)
[   55s] >       result = command()
[   55s] 
[   55s] tests/test_commands/test_inspect_venv.py:26: 
[   55s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   55s] dephell/commands/inspect_venv.py:37: in __call__
[   55s]     activate=str(venv.bin_path / shells.current.activate),
[   55s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   55s] 
[   55s] self = Shells(bin_path=PosixPath('/tmp/pytest-of-abuild/pytest-0/test_inspect_venv_command0/bin'))
[   55s] 
[   55s]     @property
[   55s]     def current(self) -> 'BaseShell':
[   55s]         shell_class = self.shells.get(self.shell_name)
[   55s]         if shell_class is None:
[   55s] >           raise LookupError('unsupported shell: {}'.format(self.shell_name))
[   55s] E           LookupError: unsupported shell: sh
[   55s] 
[   55s] /usr/lib/python3.7/site-packages/dephell_shells/_manager.py:63: LookupError
[   55s] =============================== warnings summary ===============================
[   55s] /usr/lib/python3.7/site-packages/html5lib/_trie/_base.py:3
[   55s]   /usr/lib/python3.7/site-packages/html5lib/_trie/_base.py:3: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
[   55s]     from collections import Mapping
[   55s] 
[   55s] /usr/lib/python3.7/site-packages/flatdict.py:18
[   55s]   /usr/lib/python3.7/site-packages/flatdict.py:18: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
[   55s]     class FlatDict(collections.MutableMapping):
[   55s] 
[   55s] /usr/lib/python3.7/site-packages/_pytest/mark/structures.py:325
[   55s]   /usr/lib/python3.7/site-packages/_pytest/mark/structures.py:325: PytestUnknownMarkWarning: Unknown pytest.mark.allow_hosts - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
[   55s]     PytestUnknownMarkWarning,
[   55s] 
[   55s] tests/test_actions/test_dotenv.py::test_read_dotenv[lines18-expected18]
[   55s]   /home/abuild/rpmbuild/BUILD/dephell-0.7.9/dephell/actions/_dotenv.py:38: DeprecationWarning: invalid escape sequence '\o'
[   55s]     value = decode(value, 'unicode-escape')
[   55s] 
[   55s] -- Docs: https://docs.pytest.org/en/latest/warnings.html
[   55s] ==== 1 failed, 180 passed, 39 skipped, 73 deselected, 4 warnings in 13.86s =====
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,7 +7,7 @@ before_install:
   - tree
   # install DepHell
   # https://github.com/travis-ci/travis-ci/issues/8589
-  - curl https://raw.githubusercontent.com/dephell/dephell/master/install.py | /opt/python/3.6/bin/python
+  - curl https://raw.githubusercontent.com/dephell/dephell/master/install.py | /opt/python/3.6.7/bin/python
   - dephell inspect self
 install:
   - dephell venv create --env=$ENV --python="/opt/python/$TRAVIS_PYTHON_VERSION/bin/python"
@@ -17,16 +17,16 @@ script:
 
 matrix:
   include:
-    - python: "3.5"
+    - python: "3.6.7"
       env: ENV=pytest
-    - python: "3.6"
+    - python: "3.7"
       env: ENV=pytest
-    - python: "3.7-dev"
+    - python: "3.8"
       env: ENV=pytest
     - python: "pypy3.5"
       env: ENV=pytest
 
-    - python: "3.6"
+    - python: "3.7"
       env: ENV=flake8
-    - python: "3.6"
+    - python: "3.7"
       env: ENV=typing

--- a/dephell_shells/_shells.py
+++ b/dephell_shells/_shells.py
@@ -44,6 +44,17 @@ class BashShell(BaseShell):
 
 
 @_register_shell
+class ShShell(BaseShell):
+    name = 'sh'
+    activate = 'activate'
+    interactive = True
+
+    @property
+    def command(self) -> str:
+        return '. "{}"'.format(str(self.entrypoint))
+
+
+@_register_shell
 class FishShell(BaseShell):
     name = 'fish'
     activate = 'activate.fish'

